sympy/solvers/ode/tests/test_ode.py

```diff
@@ -33,6 +33,213 @@
+            message = dsol_incorrect_msg.format(hint=our_hint, eq=eq, sol=expected_sol,dsolve_sol=dsolve_sol)
+            raise AssertionError(message)
+
+
+def _ode_solver_test(ode_examples):
+    our_hint = ode_examples['hint']
+    for example in ode_examples['examples']:
+        eq = ode_examples['examples'][example]['eq']
+        sol = ode_examples['examples'][example]['sol']
+        if our_hint not in classify_ode(eq):
+            message = hint_message.format(example=example, eq=eq, our_hint=our_hint)
+            raise AssertionError(message)
+
+        dsolve_sol = dsolve(eq,hint=our_hint)
+        if dsolve_sol not in sol:
+            message = expected_sol_message.format(example=example, eq=eq, sol=sol, dsolve_sol=dsolve_sol)
+            raise AssertionError(message)
+
+        expected_checkodesol = [(True, 0) for i in range(len(sol))]
+        if checkodesol(eq, sol) != expected_checkodesol:
+            message = checkodesol.format(example=example, eq=eq)
+            raise AssertionError(message)
+
+
+def _get_examples_ode_sol_euler_homogeneous():
+    return {
+            'hint': "nth_linear_euler_eq_homogeneous",
+            'examples':{
+                'euler_hom_01': {
+                    'eq': Eq(-3*diff(f(x), x)*x + 2*x**2*diff(f(x), x, x), 0),
+                    'sol': [Eq(f(x), C1 + C2*x**Rational(5, 2))]
+                },
+
+                'euler_hom_02': {
+                    'eq': Eq(3*f(x) - 5*diff(f(x), x)*x + 2*x**2*diff(f(x), x, x), 0),
+                    'sol': [Eq(f(x), C1*sqrt(x) + C2*x**3)]
+                },
+
+                'euler_hom_03': {
+                    'eq': Eq(4*f(x) + 5*diff(f(x), x)*x + x**2*diff(f(x), x, x), 0),
+                    'sol': [Eq(f(x), (C1 + C2*log(x))/x**2)]
+                },
+
+                'euler_hom_04': {
+                    'eq': Eq(6*f(x) - 6*diff(f(x), x)*x + 1*x**2*diff(f(x), x, x) + x**3*diff(f(x), x, x, x), 0),
+                    'sol': [Eq(f(x), C1/x**2 + C2*x + C3*x**3)]
+                },
+
+                'euler_hom_05': {
+                    'eq': Eq(-125*f(x) + 61*diff(f(x), x)*x - 12*x**2*diff(f(x), x, x) + x**3*diff(f(x), x, x, x), 0),
+                    'sol': [Eq(f(x), x**5*(C1 + C2*log(x) + C3*log(x)**2))]
+                },
+
+                'euler_hom_06': {
+                    'eq': x**2*diff(f(x), x, 2) + x*diff(f(x), x) - 9*f(x),
+                    'sol': [Eq(f(x), C1*x**-3 + C2*x**3)]
+                },
+
+                'euler_hom_07': {
+                    'eq': sin(x)*x**2*f(x).diff(x, 2) + sin(x)*x*f(x).diff(x) + sin(x)*f(x),
+                    'sol': [Eq(f(x), C1*sin(log(x)) + C2*cos(log(x)))]
+                },
+                }
+            }
+
+
+def _get_examples_ode_sol_euler_undetermined_coeff():
+    return {
+            'hint': "nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients",
+            'examples':{
+                'euler_undet_01': {
```

Comment:
I think that we can break the normal rules of indentation in this because there are so many examples and it can be repetitive and some of them lead to long lines of code. I suggest something like:
```
def _get_examples_ode_sol_euler_undetermined_coeff():
    return {
            'hint': "nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients",
            'examples':{
    'euler_undet_01': {
```
That reduces the indentation of the examples below by three levels.